subtests/docker_cli/liverestore/liverestore.py

```diff
@@ -0,0 +1,185 @@
+                            " This test will probably fail." % killmode)
+
+        self.stuff['dc'] = DockerContainers(self)
+
+    def run_once(self):
+        super(liverestore, self).run_once()
+
+        self._start_container()
+
+        # Verify that the container is running and that it's a child
+        # of the main docker daemon
+        self._verify_that_date_changes()
+        self.failif(not self._container_is_child_of_docker_daemon(),
+                    "Newly-created container is not a descendant of dockerd")
+
+        # Restart docker daemon. It should be nearly instantaneous; if it
+        # takes ~90 seconds it could be another symptom of rhbz1424709.
+        # The warning message might help someone diagnose a later failure.
+        self.stuff['dockerd_pid_orig'] = docker_daemon.pid()
+        t0 = time.time()
+        docker_daemon.restart()
+        t1 = time.time()
+        if t1 - t0 > 30:
+            self.logwarning("docker restart took %d seconds", t1 - t0)
+
+    def _start_container(self):
+        """
+        Start a container. We only need it briefly, until we (the test,
+        running in host space) can cd into its root filesystem. Container
+        will spin idly until a semaphore file is removed.
+        """
+        c_name = self.stuff['dc'].get_unique_name()
+
+        fin = DockerImage.full_name_from_defaults(self.config)
+        subargs = ['--detach', '--name=' + c_name, fin,
+                   'bash -c "echo READY;'
+                   ' while :; do date +%s; sleep 1;done"']
+        dkrcmd = AsyncDockerCmd(self, 'run', subargs)
+        dkrcmd.execute()
+        dkrcmd.wait_for_ready(c_name)
+        self.stuff['container_name'] = c_name
+
+    def _verify_that_date_changes(self):
+        """
+        Verify that container process is running, by running 'docker log'
+        on it sequentially and making sure that (1) we get new output,
+        and (2) the new output corresponds to current time.
+        """
+        log_0 = self._container_log()
+
+        # Wait up to 5 seconds for output to change
+        end_at = time.time() + 5
+        while time.time() <= end_at:
+            log_1 = self._container_log()
+            if log_1 != log_0:
+                break
+            time.sleep(1)
+
+        self.failif(len(log_0) == len(log_1),
+                    "Container log line count did not grow")
+        container_time = int(log_1[-1])
+        self.failif(abs(time.time() - container_time) > 2,
+                    "Last-output time from container (%s) is >2 seconds"
+                    " off from system time" % time.ctime(container_time))
+
+    def _container_log(self):
+        """
+        Returns container log as a list of strings
+        """
+        c_name = self.stuff['container_name']
+        result = mustpass(DockerCmd(self, "logs", [c_name]).execute())
+        return result.stdout.strip().split("\n")
```

Comment:
The kernel (and even the VM's host) [clocks are non-monotonic](https://en.wikipedia.org/wiki/Monotonic_function#See_also).  There's no guarantee that can't flow backward or repeat jiffies in totally unpredictable ways.  Consider if ``NTP`` or ``chrony`` happens to tweak backwards a few milliseconds, or changes time-zones immediately before a second's boundry.

The simple fix here is to make the time checks both ordered and interval-based (instead of absolute).  So, have it dump the value of a counter that increments on 100ms intervals (``sleep 0.1s`` for example).  Otherwise this test could fail in an unpredictable and extremely difficult to diagnose way.